### PR TITLE
Add getValidThreadColors

### PR DESCRIPTION
**Pull request by AstroCB**
Adds a function to get accepted values for `changeThreadColor` and validation within `changeThreadColor` to prevent undefined behavior for non-supported colors as per #506.

### CODE REVIEW

**Comment by Schmavery** [comment]
Thanks for contributing some more changes :)
Is there any reason you're not just exposing the data directly? As in:
```
module.exports = function(defaultFuncs, api, ctx) {
    return ["#44bec7", "#ffc300", "#fa3c4c", "#d696bb", "#6699cc", ...];
};
```

I'm also wondering if it wouldn't be more readable if we actually made it a map of the human-readable names exposed by facebook to the colors. The python clone of this api [does this](https://github.com/carpedm20/fbchat/blob/master/fbchat/models.py#L185). i.e.
```
{
    MessengerBlue = ''
    Viking = '#44bec7'
    GoldenPoppy = '#ffc300'
    RadicalRed = '#fa3c4c'
    Shocking = '#d696bb
    ...
}
```
Might end up with more readable code and make it more possible for us to add/adapt/remove colors as time goes on (though I'll admit the names are... strange at best).

**Comment by AstroCB** [comment]
Ah, I haven't seen any constant (non-functional) data exposed via the API so I wasn't sure if that would be against the design you guys have set up.

Are those names the official ones exposed by Facebook? I haven't seen them anywhere.

**Comment by AstroCB** [comment]
I switched it to a dictionary using those names; the code should definitely be more readable that way.

**Comment by bsansouci** [comment]
Seems good to me though I'd change the docs to say that it is an object rather than it returns an object.

File: DOCS.md [hunk 1]
```diff
@@ -715,6 +716,29 @@ login({appState: JSON.parse(fs.readFileSync('appstate.json', 'utf8'))}, (err, ap
 
 ---------------------------------------
 
+<a name="validThreadColors"></a>
+### api.validThreadColors
+
+Returns a dictionary mapping names of all currently valid thread colors to their hexadecimal values that are accepted by [`api.changeThreadColor`](#changeThreadColor). These colors, listed below, are the ones present in the palette UI used for selecting thread colors on the Messenger client. Due to Facebook backend changes, the thread color can no longer be set to an arbitrary hex value.
```
Comment (Schmavery) [inline DOCS.md]: Maybe we can move the note about "Due to Facebook backend changes, the thread color can no longer be set to an arbitrary hex value." to the CHANGELOG.md file and put a link there to this PR?  (See the other entries for an example)

**Comment by Schmavery** [comment]
I feel like the validity is implied by the fact that we're providing the colors, so just `threadColors` is probably more concise.  Thanks!

**Comment by AstroCB** [comment]
Done.

**Comment by Schmavery** [comment]
Seems good, thanks a lot.

**Pull request merged**